Makefile

```diff
@@ -29,10 +29,28 @@ GIT_SUBMODULE_UPDATE := git submodule update --init --recursive
 
 else # "variables.mk" was included. Business as usual until the end of this file.
 
-ifeq ($(OS),Windows_NT)     # is Windows_NT on XP, 2000, 7, Vista, 10...
- detected_OS := Windows
+# Determine the OS
+detected_OS := $(shell uname -s)
+ifeq ($(detected_OS),Darwin)
+  detected_OS := Darwin
+else ifeq ($(detected_OS),Linux)
+  detected_OS := Linux
+else ifneq (,$(findstring MINGW,$(detected_OS)))
```

Comment:
in windows systems, the OS variable is usually empty by default, so I use the `uname` command for information. Since MINGW is very common in windows, use that idea to differentiate windows and non-windows. but yeah i got your point, i will try make more simplified.